pysiaf/utils/tools2.py

```diff
@@ -0,0 +1,217 @@
+
+def matchV2V3(apName1, apName2):
+    """ Use the V2V3 from aperture named apName1 in aperture apName2, modifying XDetRef, YDetRef,
+    XSciRef YSciRef to match
+    Also  copy or shift the polynomial coefficients to reflect the new reference point origin
+    and for NIRCam recalculate angles. """
+
+    instruments = {'NRC':'NIRCam', 'NRS':'NIRSpec', 'FGS':'FGS', 'MIR':'MIRI', 'NIS':'NIRISS'}
+    insCode = apName1[:3]
+    ins = instruments[insCode]
+    print ('\n******************************************************')
+    print ('Instrument', ins)
+    siaf = Siaf(ins)
+    ap1 = siaf.apertures[apName1]
+    ap2 = siaf.apertures[apName2]
+    assert (ap2.AperType in ['FULLSCA', 'SUBARRAY', 'ROI']), "2nd aperture must be pixel-based"
+    order = ap1.Sci2IdlDeg
+    V2Ref1 = ap1.V2Ref
+    V3Ref1 = ap1.V3Ref
+    newV2Ref = V2Ref1 # in all cases
+    newV3Ref = V3Ref1
+    print ('Current Vref', ap2.V2Ref, ap2.V3Ref)
+    print ('Shift to    ', V2Ref1, V3Ref1)
+
+    # Detector and Science axes may go in opposite directions
+    ySign = cos(radians(ap2.DetSciYAngle))
+    xSign = ap2.DetSciParity * ySign
+
+
+    if apName1[:5] == apName2[:5]: # Common detector - most things copied
+        print ('Detector ', apName1[:5])
+        newXDetRef = ap1.XDetRef
+        newYDetRef = ap1.YDetRef
+        newXSciRef = ap2.XSciRef + xSign*(newXDetRef-ap2.XDetRef) 
+        newYSciRef = ap2.YSciRef + ySign*(newYDetRef-ap2.YDetRef)
+        newV3SciXAngle = ap1.V3SciXAngle
+        newV3SciYAngle = ap1.V3SciYAngle
+        newV3IdlYAngle = ap1.V3IdlYAngle
+        newA = get_polynomial_coefficients(ap1, 'Sci2IdlX')
+        newB = get_polynomial_coefficients(ap1, 'Sci2IdlY')
+        newC = get_polynomial_coefficients(ap1, 'Idl2SciX')
+        newD = get_polynomial_coefficients(ap1, 'Idl2SciY')
+        print ('{:10.4f} {:10.4f} {:10.4f} {:10.4f} {:10.4f} {:10.4f} {:10.4f} {:10.4f} {:10.4f}'.format(ap2.V2Ref, ap2.V3Ref, ap2.XDetRef, ap2.YDetRef,
+                                                                                       ap2.XSciRef,ap2.YSciRef, ap2.V3SciXAngle, ap2.V3SciYAngle, ap2.V3IdlYAngle))
+        print ('{:10.4f} {:10.4f} {:10.4f} {:10.4f} {:10.4f} {:10.4f} {:10.4f} {:10.4f} {:10.4f} '.format(newV2Ref, newV3Ref, newXDetRef, newYDetRef,
+                                                                                        newXSciRef,newYSciRef, newV3SciXAngle, newV3SciYAngle, newV3IdlYAngle))
+    else:
+        # Need to work in aperture 2 Ideal coordinate system
+        print ('Detector 1', ap1.AperName[:5], '  Detector 2', ap2.AperName[:5])
+        V2Ref2 = ap2.V2Ref
+        V3Ref2 = ap2.V3Ref
+        theta0 = ap2.V3IdlYAngle
+        print ('Initial VRef', V2Ref2, V3Ref2)
+        print ('Initial theta', theta0)
+        theta = radians(theta0)
+        # Express reference point 1 V2V3  in Ideal2 coordinates
+        XIdl2 = ap2.VIdlParity*((V2Ref1-V2Ref2)*cos(theta) - (V3Ref1-V3Ref2)*sin(theta))
```

Comment:
I believe this operation can be accomplished by using something like
`x_idl_2, y_idl_2 = aperture_2.tel_to_idl(v2_ref_1, v2_ref_2)`